test/hotspot/jtreg/serviceability/jvmti/vthread/SelfSuspendDisablerTest/SelfSuspendDisablerTest.java

```diff
@@ -51,26 +57,57 @@ private static void sleep(long millis) {
         }
     }
 
+    private static void testJvmtiThreadState(Thread thread, int expectedState) throws RuntimeException {
```

Comment:
Minor nit, "throws RuntimeException" is not needed here.

Reply:
Ah, yes. I was thinking it is not needed but forgot to double-check.
Fixed now.

> The fix looks good but I'm wondering why the SelfSuspendDisabledTest is being used to test this bug. I guess I expected to see a test for GetThreadState instead.

The GetThreadState by its instrumental role can be used by many tests.
My initial intent was to check the terminated virtual thread case.
It looked as unneeded overhead to create a new test for this.
So, I've found this small test which already has convenient infrastructure to recreate needed conditions.
Then I decided to extend GetThreadState coverage in this test a little bit.
I can create a GetThreadState specific test if you think it is worth it.
 Interesting enough that I've found the existing test which already had needed coverage:
   ` test/hotspot/jtreg/serviceability/jvmti/thread/GetThreadState/thrstat03`
But this test was adjusted to adopt to incorrect GetThreadState result for virtual threads, so I had to fix it now.
Strongly speaking, the update of SelfSuspendDisabledTest is not needed.
But I feel it is worth to keep it.

Reply:
> Interesting enough that I've found the existing test which already had needed coverage: ` test/hotspot/jtreg/serviceability/jvmti/thread/GetThreadState/thrstat03` But this test was adjusted to adopt to incorrect GetThreadState result for virtual threads, so I had to fix it now. Strongly speaking, the update of SelfSuspendDisabledTest is not needed. But I feel it is worth to keep it.

This goes to my surprise that we didn't have a test already but you've found it, and found that it should have caught this bug except that it has been changed. Good to find this and I that is the right place to have a unit test for GetThreadState.